tools/update-eslint.sh

```diff
@@ -4,24 +4,18 @@
 
 # Depends on npm, npx, and node being in $PATH.
 
-# This script must be be in the tools directory when it runs because it uses
-# $BASH_SOURCE[0] to determine directories to work in.
+# This script must be in the tools directory when it runs because it uses the
+# script source file path to determine directories to work in.
 
 cd "$( dirname "$0" )" || exit
-rm -rf node_modules/eslint
+rm -rf node_modules/eslint node_modules/eslint-plugin-markdown
 (
     mkdir eslint-tmp
     cd eslint-tmp || exit
     npm init --yes
 
     npm install --global-style --no-bin-links --production --no-package-lock eslint@latest
-
-    (
-        cd node_modules/eslint || exit
-
-        npm install --no-bin-links --production --no-package-lock eslint-plugin-markdown@latest
-    )
-
+    npm install --global-style --no-bin-links --production --no-package-lock eslint-plugin-markdown@latest
 
     # Use dmn to remove some unneeded files.
     npx dmn@2.2.2 -f clean
```

Comment:
I think it will not work if this is backported? Not sure if we want to do that, probably not.

Reply:
Wouldn't that break anyone trying to use npm 6?

Reply:
That's what I was thinking, it has nothing to do with backporting :)

Reply:
what? is it possible that folks running this script have npm6 as their global installed npm version? 🤔 that sounds risky... in that case I would advocate to replace all **npm** usages it with the current shipped npm version, it's very important to ensure the script is run always with the same npm release line to avoid churn and edge case problems that might occur.

I can float a PR to shift that across these `./tools/update-*` scripts, I imagine something like:

```patch
+NPMCLI="$(pwd)/deps/npm"
+
-npm init -y
+node $NPMCLI init -y
```

Reply:
The diff might be hard to review if the script is run from two different npm release lines but what else could go wrong? Everything should work as long as the module resolution algorithm does not change no?

Also there is no guarantee that two different npm version on the release same line will produce the same output, right?

Reply:
> The diff might be hard to review if the script is run from two different npm release lines but what else could go wrong? Everything should work as long as the module resolution algorithm does not change, no?

well yeah, the default resolution algo does change and that's what broke the current workflow at first, having an `ERESOLVE` error popping up at the origin of the issue the current PR is fixing 😊 

> Also there is no guarantee that two different npm versions on the same release line will produce the same output, right?

technically no, since bug fixes in the resolution algo could theoretically yield a different (fixed) install tree but in theory npm7 installs are deterministic and don't suffer from race-condition issues the same way npm6 did.

Reply:
I'll mark this specific conversation as resolved since it's not relevant to this PR 😄 I'll make sure to ping both of you if I send a PR changing that. Thanks!